source/main.cpp

```diff
@@ -67,18 +86,22 @@ int main(int argc, char** argv)
 
     sh3_gl::program::load_error err;
     sh3_gl::program test("test", err);
-    sh3_gl::vao<MyAttributes> myVao;
 
-    myVao.Bind();
-    sh3_gl::buffer_object& verts = myVao[MyAttributes::VERTEX_SLOT];
+    using Triangle = sh3_gl::vao<TriangleAttributes>;
+
+    Triangle triVao;
+
+    triVao.Bind();
+
+    sh3_gl::buffer_object& verts = triVao[Triangle::Slot::VERTEX];
     verts.BufferData(g_vertex_buffer_data, sizeof(g_vertex_buffer_data), GL_STATIC_DRAW);
-    myVao.SetDataLocation(MyAttributes::VERTEX_SLOT, sh3_gl::vao<MyAttributes>::DataType::FLOAT, 3, 0, 0);
+    triVao.SetDataLocation(Triangle::Slot::VERTEX, Triangle::DataType::FLOAT, 3, 0, 0);
```

Comment:
You're using words from different concepts that confuse my brain. I think the answer is "yes" though.

The `Target` for `triVao[slot]` is `TriangleAttributes::Targets[slot]`.
So if we had `Targets = { Target::ARRAY_BUFFER, Target::TEXTURE_BUFFER }`, then the vertex buffer would be `ARRAY_BUFFER` and the color buffer would be `TEXTURE_BUFFER`.